specification/msi/ManagedIdentity.Management/models.tsp

```diff
@@ -0,0 +1,277 @@
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: added in TypeSpec migration, follow aka.ms/tsp/conversion-fix for details"
+@summary("User Assigned Identity properties.")
+model UserAssignedIdentityProperties {
+  /**
+   * The id of the tenant which the identity belongs to.
+   */
+  @visibility(Lifecycle.Read)
+  tenantId?: Azure.Core.uuid;
+
+  /**
+   * The id of the service principal object associated with the created identity.
+   */
+  @visibility(Lifecycle.Read)
+  principalId?: Azure.Core.uuid;
+
+  /**
+   * The id of the app associated with the identity. This is a random generated UUID by MSI.
+   */
+  @visibility(Lifecycle.Read)
+  clientId?: Azure.Core.uuid;
+
+  /**
+   * Enum to configure regional restrictions on identity assignment, as necessary.
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create, Lifecycle.Update)
+  isolationScope?: IsolationScope;
+}
+
+/**
+ * Describes an identity resource.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: added in TypeSpec migration, follow aka.ms/tsp/conversion-fix for details"
+model IdentityUpdate extends Azure.ResourceManager.Foundations.Resource {
+  /**
+   * The geo-location where the resource lives
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  location?: string;
+
+  /**
+   * Resource tags
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: added in TypeSpec migration, follow aka.ms/tsp/conversion-fix for details"
+  @visibility(Lifecycle.Read, Lifecycle.Create, Lifecycle.Update)
+  tags?: Record<string>;
+
+  /**
+   * The properties associated with the identity.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-private-usage" "FIXME: added in TypeSpec migration, follow aka.ms/tsp/conversion-fix for details"
+  @Azure.ResourceManager.Private.conditionalClientFlatten
+  properties?: UserAssignedIdentityProperties;
+}
+
+/**
+ * Values returned by the List operation for federated identity credentials.
+ */
+model FederatedIdentityCredentialsListResult
+  is Page<FederatedIdentityCredential>;
+@@extension(FederatedIdentityCredentialsListResult.value,
+  "x-ms-identifiers",
+  #["id"]
+);
+
+/**
+ * The properties associated with a federated identity credential.
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "FIXME: added in TypeSpec migration, follow aka.ms/tsp/conversion-fix for details"
+@summary("Federated identity credential properties.")
+model FederatedIdentityCredentialProperties {
+  /**
+   * The URL of the issuer to be trusted.
+   */
+  issuer: url;
+
+  /**
+   * The identifier of the external identity.
+   */
+  subject: string;
+
+  /**
+   * The list of audiences that can appear in the issued token.
+   */
+  audiences: string[];
+}
+
+/**
+ * The default resourceUri parameter type.
+ */
+@doc("The default resourceUri parameter type.")
+model ScopeParameter {
```

Comment:
I think we don't need this ScopeParameter any more?

Reply:
removed